packages/backend/src/ee/services/ai/tools/findExplores.ts

```diff
@@ -1,51 +1,145 @@
+        .filter((field) => allTableNames.includes(field.tableName))
+        .sort((a, b) => (b.chartUsage ?? 0) - (a.chartUsage ?? 0));
+
+    return `
+<Explore id="${table.name}">
+    <Label>${table.label}</Label>
+    <BaseTable alt="ID of the base table">${table.name}</BaseTable>
+
+    ${
+        table.aiHints && table.aiHints.length > 0
+            ? `
+    <AI Hints>
+        ${table.aiHints.map((hint) => `<Hint>${hint}</Hint>`).join('\n')}
+    </AI Hints>`.trim()
+            : ''
+    }
+
+    <Description alt="Description of the base table">
+${truncate(table.description, {
+    length: descriptionMaxLength,
+    omission: '... (truncated, read more by calling get explore tool)',
+})}
+    </Description>
+
+    ${
+        table.joinedTables && table.joinedTables.length > 0
+            ? `
+    <JoinedTables alt="IDs of the joined tables" size="${
+        table.joinedTables.length
+    }">
+        ${table.joinedTables
+            .map(
+                (joinedTable) =>
+                    `<JoinedTable id="${joinedTable.table}">${joinedTable.table}</JoinedTable>`,
+            )
+            .join('\n\n')}
+    </JoinedTables>
+`.trim()
+            : ''
+    }
+
+    <Fields alt="All fields from all tables" size="${
+        dimensions.length + metrics.length
+    }">
+        <Dimensions alt="dimension ids" size="${dimensions.length}">
+            ${dimensions
+                .map(
+                    (f) =>
+                        `<Dimension fieldId="${getItemId({
+                            name: f.name,
+                            table: f.tableName,
+                        })}">${f.label}</Dimension>`,
+                )
+                .join('\n')}
+        </Dimensions>
+
+        <Metrics alt="metric ids" size="${metrics.length}">
+            ${metrics
+                .map(
+                    (metric) =>
+                        `<Metric fieldId="${getItemId({
+                            name: metric.name,
+                            table: metric.tableName,
+                        })}" popularity="${metric.chartUsage ?? 0}">${
+                            metric.label
+                        }</Metric>`,
+                )
+                .join('\n')}
+        </Metrics>
+    </Fields>
+</Explore>`.trim();
+};
+
+export const getFindExplores = ({ findExplores }: Dependencies) => {
+    const schema = z.object({
+        page: z
+            .number()
+            .nullable()
+            .describe(
+                'Use this to paginate through the results. Starts at 1 and increments by 1.',
+            ),
+    });
 
     return tool({
-        description: `Use this tool to get information about Explores (models) available to you.`,
+        description: `Use this tool to list available Explores with joined tables with hints, description, and available fields. The list is paginated, you can paginate through the results by recalling the tool with the next page token.`,
```

Comment:
should this mention that the dimensions and metrics are ordered by popularity metric = chart usage?

Reply:
good point, updated